README.adoc

```diff
@@ -75,88 +61,156 @@ It could be omitted as the title already says that it is an experimental feature
+|===
 
-⛔Example (contains a possible solution):
-      The provided label is not in the database.
+The following are guidelines on how to write each of the notification components.
 
-✅Example:
-     The label `Perso` does not exist.
 
+==== StatusCode syntax
 
+All Neo4j status codes follow the syntax `Neo.[Class].[Subclass].[Name]`.
 
-=== Description
+[options="header",cols="h,1,2,3,3,2",]
+|===
+| StatusCode
+| `Neo`
+| `[Class]`
+| `[Subclass]`
+| `[Name]`
+| Example
+
+| Notification
+| `Neo`
+| `ClientNotification`
+| One of `Statement`, `Procedure`, `Schema`, `Database`, `Security`, or `Request`
+| Should be specific and explains what this code refers to, e.g., `ExhaustiveShortestPath`.
+| `Neo.ClientNotification.Statement.ExhaustiveShortestPath`
+
+| Error
+| `Neo`
+| One of `ClientError`, `TransientError`, `DatabaseError`
+| One of `ChangeDataCapture`, `Cluster`, `Database`, `Fabric` (deprecated), `General`, `Procedure`, `Request`, `Routing`, `Schema`, `Security`, `Statement`, `Transaction`.
+| Should be specific and explains what this code refers to, e.g., `RemoteExecutionFailed`.
+| `Neo.DatabaseError.Statement.RemoteExecutionFailed`
+|===
 
-This is the description of the returned code.
 
-The description should contain the following:
+==== Notification title/Error description
 
-* What caused the notification to be returned
-* Why it might be a problem - what the consequences are
-* If possible, a simple solution, otherwise, a reference to the docs using the sentence:
-_See Status Codes documentation for suggestions for improvement._
+The notification title and the error description contain similar information.
+Therefore, they follow the same guidelines.
+Both should be brief (one short sentence), specific, and inform on the current situation and what code of the query triggered the notification/error (add it within backticks to show that it’s a code snippet).
 
+* Do not use exclamation marks.
+* Use sentence case (you capitalize the first letter, proper nouns, and names, just like you would in a normal sentence) and punctuation.
+* Avoid explaining the cause, consequences, or solution.
 
-⛔Example:
-Using shortest path with an unbounded pattern will likely result in long execution times.
-It is recommended to use an upper limit to the number of node hops in your pattern.
+.Examples for notification title and error description
+[options="header",cols="h,2,2",]
+|===
+| Example
+| ⛔ Old title/description
+| ✅ New title/description
+
+| Notification title 1
+| The provided pattern is unbounded, consider adding an upper limit to the number of node hops.
+| The pattern `<pattern>` is unbounded.
+
+| Notification title 2
+| The provided label is not in the database.
+| The label `Perso` does not exist.
+
+| Notification title 3
+| This feature is deprecated and will be removed in future versions.
+| (Deprecated) The repeated variable length relationship `r` is bound more than once.
```

Comment:
I think we should remove this example as all deprecations have the above title and that cannot be changed (it is breaking to change in 5.x and the title will anyway be removed in 6.x). The others cannot be changed either, but can more be seen as general examples